### PR TITLE
feat: add tarball download for zlib

### PR DESCRIPTION
**Pull request by j8xixo12**
I added zlib build script tarball download support

### CODE REVIEW

File: scripts/build.d/zlib [hunk 1]
```diff
@@ -2,11 +2,36 @@
...
+SRCSYNC=${SRCSYNC:-tarball}
 
-syncgit https://github.com/madler ${pkgname} ${pkgbranch}
+if [ "${SRCSYNC}" == "tarball" ]; then
+
+  pkgname=zlib
+  pkgver=${VERSION:-v1.2.13}
+  pkgfull=${pkgname}-${pkgver:1}
+  pkgfn=${pkgfull}.tar.gz
+  pkgurl=https://github.com/madler/zlib/archive/refs/tags/${pkgver}.tar.gz
```
Comment (j8xixo12) [inline scripts/build.d/zlib:13]: There are 2 url can be used
1. Release tarball
2. Tag tarball

Release tarball only has 1.2.13, but Tag tarball has many versions, therefore I choose Tag tarball as build script download url

File: scripts/build.d/zlib [hunk 2]
```diff
@@ -16,5 +41,4 @@ pushd ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src/${pkgfull} > /dev/null
   buildcmd install.log make install
 
 popd > /dev/null
-
```
Comment (yungyuc) [inline scripts/build.d/zlib:19]: Probably you do not need to change this line.

**Review by yungyuc — APPROVED** [review]
LGTM.  Thanks!

**Comment by yungyuc** [comment]
Fixes #150

**Pull request merged**